### PR TITLE
Tools: autotest: allow autotest to connect directly to SITL

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by magicrub** [comment]
would this allow us to push up the sim speed? It will hopefully at least give us more stable comms (and thus repeatable CI result) at whatever high-speed we're running now.

**Comment by peterbarker** [comment]
@magicrub yes, it should allow us to push the speeds up a bit.

File: Tools/autotest/rover.py [hunk 1]
```diff
@@ -1155,23 +1161,25 @@ def test_sysid_enforce(self):
...
+            success = None
             try:
-                self.mav.mav.srcSystem = 243
+                # temporarily set a different system ID than normal:
+                self.mav.mav.srcSystem = 72
                 self.arm_vehicle(timeout=5)
                 self.disarm_vehicle()
-                success = False
-            except AutoTestTimeoutException as e:
+                sucess = False
```
Comment (khancyr) [inline Tools/autotest/rover.py:1174]: sucess ?

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -4855,6 +4852,8 @@ def rc_defaults(self):
 
     def test_manual_control(self):
         '''test manual_control mavlink message'''
+        self.set_parameter("SYSID_MYGCS", self.mav.source_system)
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply Tools/autotest/arducopter.py:4855]: Could be.  OTOH, doing it here keeps everything explicit. and doesn't waste time for the 99% of tests that don't need it :-)

File: Tools/autotest/rover.py (hunk 1 above)
Comment (peterbarker) [reply Tools/autotest/rover.py]: Fixed.

File: Tools/autotest/common.py [hunk 3]
```diff
@@ -5118,10 +5250,11 @@ def init(self):
         self.progress("Starting simulator")
         self.start_SITL()
 
-        self.start_mavproxy()
+        os.environ['MAVLINK20'] = '1'
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply Tools/autotest/common.py:5253]: I didn't add it for fun :-)  Just added it when things died *horribly*.

Removed and ran it on Rover, first failure is this:

```
  RCOverrides (Test RC overrides) (rc_channels_override_send() takes from 11 to 12 positional arguments but 21 were given) (see /home/pbarker/rc/buildlogs/Rover-RCOverrides.txt)
```

So yeah, I think it stays :-)

**Review by peterbarker — COMMENTED** [review]
Thanks for the review @khancyr

File: Tools/autotest/common.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (khancyr) [reply Tools/autotest/common.py:5253]: Humm maybe it is time to update mavutil to default mavlink2 or at least load the v20 messages instead of v10

**Pull request merged**